kornia/geometry/camera/distortion_kannala_brandt.py

```diff
@@ -0,0 +1,245 @@
+    rth2 = un**2 + vn**2
+
+    # TODO: explore stop condition (won't work with pytorch with batched inputs)
+    # Additionally, with this stop condition we can avoid adding 1e-8 to the denominator
+    # in the return statement of the function.
+
+    # if rth2.abs() < 1e-8:
+    #     return distorted_points_in_camera
+
+    rth = rth2.sqrt()
+
+    th = rth.sqrt()
+
+    iters = 0
+
+    # gauss-newton
+
+    while True:
+        th2 = th**2
+        th4 = th2**2
+        th6 = th2 * th4
+        th8 = th4**2
+
+        thd = th * (1.0 + k0 * th2 + k1 * th4 + k2 * th6 + k3 * th8)
+
+        d_thd_wtr_th = 1.0 + 3.0 * k0 * th2 + 5.0 * k1 * th4 + 7.0 * k2 * th6 + 9.0 * k3 * th8
+
+        step = (thd - rth) / d_thd_wtr_th
+        th = th - step
+
+        iters += 1
+
+        # TODO: improve stop condition by masking only the elements that have converged
+        th_abs_mask = th.abs() < 1e-8
+
+        if th_abs_mask.all():
+            break
+
+        if iters >= 20:
+            break
+
+    radius_undistorted = th.tan()
+
+    undistorted_points = ops.where(
+        radius_undistorted[..., None] < 0.0,
+        ops.stack(
+            [
+                -radius_undistorted * un / (rth + 1e-8),
+                -radius_undistorted * vn / (rth + 1e-8),
+            ],
+            dim=-1,
+        ),
+        ops.stack(
+            [
+                radius_undistorted * un / (rth + 1e-8),
+                radius_undistorted * vn / (rth + 1e-8),
+            ],
+            dim=-1,
+        ),
+    )
+
+    return undistorted_points
+
+
+def dx_distort_points_kannala_brandt(projected_points_in_camera_z1_plane: Tensor, params: Tensor) -> Tensor:
+    r"""Compute the derivative of the x distortion with respect to the x coordinate.
+
+    .. math::
+        \frac{\partial u}{\partial x} =
+        \begin{bmatrix} f_x & 0 \\ 0 & f_y \end{bmatrix}
+
+    Args:
+        projected_points_in_camera_z1_plane: Tensor representing the points to distort with shape (..., 2).
+        params: Tensor representing the parameters of the Kannala-Brandt distortion model with shape (..., 8).
+
+    Returns:
+        Tensor representing the derivative of the x distortion with respect to the x coordinate with shape (..., 2).
+
+    Example:
+        >>> points = torch.tensor([1., 2.])
+        >>> params = torch.tensor([1000.0, 1000.0, 320.0, 280.0, 0.1, 0.01, 0.001, 0.0001])
+        >>> dx_distort_points_kannala_brandt(points, params)
+        tensor([[ 486.0507, -213.5573],
+                [-213.5573,  165.7147]])
+    """
+    KORNIA_CHECK_SHAPE(projected_points_in_camera_z1_plane, ["*", "2"])
+    KORNIA_CHECK_SHAPE(params, ["*", "8"])
+
+    a = projected_points_in_camera_z1_plane[..., 0]
+    b = projected_points_in_camera_z1_plane[..., 1]
+
+    fx, fy = params[..., 0], params[..., 1]
+
+    k0 = params[..., 4]
+    k1 = params[..., 5]
+    k2 = params[..., 6]
+    k3 = params[..., 7]
+
+    # TODO: return identity matrix if a and b are zero
```

Comment:
maybe add it now?

Reply:
this one is tricky because we cannot return when the condition is hitted only by one sample

Reply:
ouch, i see